lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -472,6 +475,30 @@ bool exceeds64Bits(const Inst *I, std::set<const Inst *> &Visited) {
   return false;
 }
 
+bool canDifferInLSB(InstContext &IC, BlockPCs &BPCs,
+                    std::vector<InstMapping> &PCs,
+                    InstMapping &M, unsigned Timeout,
+                    SMTLIBSolver *SMTSolver) {
+  Inst *LHSOne = IC.getConst(llvm::APInt(M.LHS->Width, 1));
+  Inst *NewLHS = IC.getInst(Inst::And, M.LHS->Width, {M.LHS, LHSOne});
+  Inst *RHSOne = IC.getConst(llvm::APInt(M.RHS->Width, 1));
+  Inst *NewRHS = IC.getInst(Inst::And, M.RHS->Width, {M.RHS, RHSOne});
+  // TODO: Experiment with larger masks: 3, 7, MSB, etc.
+
+  InstMapping NewMapping{NewLHS, NewRHS};
+
+  auto Query = BuildQuery(IC, BPCs, PCs, NewMapping, 0);
+
+  bool QueryIsSat;
+  auto EC = SMTSolver->isSatisfiable(Query, QueryIsSat, 0, 0, Timeout);
+  if (EC) {
+    if (DebugLevel > 1)
+      llvm::errs() << "error!\n";
```

Comment:
should say something more interesting here, and probably at all debug levels

Reply:
Expanded the error message, note that just "error!\n" is used as an error message in multiple places in  this file.
As for all debug levels: probably not necessary because if the query fails, nothing bad happens; it just fails to prune early.